zarr/meta.py

```diff
@@ -138,16 +138,20 @@ def decode_fill_value(v, dtype):
             return np.NINF
         else:
             return np.array(v, dtype=dtype)[()]
-    elif dtype.kind in 'SV':
+    elif dtype.kind == 'S':
         # noinspection PyBroadException
         try:
             v = base64.standard_b64decode(v)
-            v = np.array(v, dtype=dtype)[()]
-            return v
         except Exception:
             # be lenient, allow for other values that may have been used before base64
             # encoding and may work as fill values, e.g., the number 0
-            return v
+            pass
+        v = np.array(v, dtype=dtype)[()]
+        return v
+    elif dtype.kind == 'V':
+        v = base64.standard_b64decode(v)
+        v = np.array(v, dtype=dtype.str).view(dtype)[()]
```

Comment:
Clever! Are we doing any round trip tests of encoding/decoding and vice versa?

Reply:
[This test](https://github.com/zarr-developers/zarr/blob/b113d0aa8295dab550fc54a2d4b16b49d5a61b23/zarr/tests/test_meta.py#L67) covers encoding/decoding of metadata for a structured array. Probably could add a specific test for encoding/decoding structured array fill value.

Reply:
Looks good. Thanks for the pointer.

It couldn't hurt to have such a test if you want to add it. However it seems like this is already being tested by the array round trip. So don't think it is strictly necessary to test fill values specifically.